src/tests/ftest/util/general_utils.py

```diff
@@ -90,7 +101,205 @@ def process_host_list(hoststr):
+        exclude (list|None): list of strings to not return
+
+    Returns:
+        str: a string of random ascii letters and numbers
+
     """
-    DAOS API exception class
+    exclude = exclude if isinstance(exclude, list) else []
+    random_string = None
+    while not isinstance(random_string, str) or random_string in exclude:
+        random_string = "".join(
+            random.choice(string.ascii_uppercase + string.digits)
+            for _ in range(length))
+    return random_string
+
+
+@fail_on(DaosApiError)
+def get_pool(context, mode, size, name, svcn=1, log=None):
+    """Return a DAOS pool that has been created an connected.
+
+    Args:
+        context (DaosContext): the context to use to create the pool
+        mode (int): the pool mode
+        size (int): the size of the pool
+        name (str): the name of the pool
+        svcn (int): the number of pool replica leaders
+        log (DaosLog|None): object for logging messages
+
+    Returns:
+        DaosPool: an object representing a DAOS pool
+
+    """
+    if log:
+        log.info("Creating a pool")
+    pool = DaosPool(context)
+    pool.create(mode, os.geteuid(), os.getegid(), size, name, svcn=svcn)
+    if log:
+        log.info("Connecting to the pool")
+    pool.connect(1 << 1)
+    return pool
+
+
+@fail_on(DaosApiError)
+def get_container(context, pool, log=None):
+    """Retrun a DAOS a container that has been created an opened.
+
+    Args:
+        context (DaosContext): the context to use to create the container
+        pool (DaosPool): pool in which to create the container
+        log (DaosLog|None): object for logging messages
+
+    Returns:
+        DaosContainer: an object representing a DAOS container
+
+    """
+    if log:
+        log.info("Creating a container")
+    container = DaosContainer(context)
+    container.create(pool.handle)
+    if log:
+        log.info("Opening a container")
+    container.open()
+    return container
+
+
+@fail_on(DaosApiError)
+def kill_server(server_group, context, rank, pool, log=None):
+    """Kill a specific server rank.
+
+    Args:
+        server_group (str):
+        context (DaosContext): the context to use to create the DaosServer
+        rank (int): daos server rank to kill
+        pool (DaosPool): the DaosPool from which to exclude the rank
+        log (DaosLog|None): object for logging messages
+
+    Returns:
+        None
+
+    """
+    if log:
+        log.info("Killing DAOS server {} (rank {})".format(server_group, rank))
+    server = DaosServer(context, server_group, rank)
+    server.kill(1)
+    if log:
+        log.info("Excluding server rank {}".format(rank))
+    pool.exclude([rank])
+
+
+@fail_on(DaosApiError)
+def get_pool_status(pool, log):
+    """Determine if the pool rebuild is complete.
+
+    Args:
+        pool (DaosPool): pool for which to determine if rebuild is complete
+        log (logging): logging object used to report the pool status
+
+    Returns:
+        None
```

Comment:
This function returns the return value of the pool.pool_query() function - need to update comment.